utils/swift_build_support/swift_build_support/toolchain.py

```diff
@@ -0,0 +1,124 @@
+# -*- python -*-
+# swift_build_support/toolchain.py - Detect host machine's versioned executables
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See http://swift.org/LICENSE.txt for license information
+# See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+#
+# ----------------------------------------------------------------------------
+#
+# Find the path to a Clang executable on the host machine that is most
+# suitable for building Swift.
+#
+# ----------------------------------------------------------------------------
+
+from __future__ import absolute_import
+
+import collections
+import platform
+import subprocess
+
+from . import xcrun
+from .which import which
+
+
+class Toolchain(object):
+    """
+    A class that stores accessible paths to system commands.
+
+    It requires a 'cc' and 'cxx' tool in its keyword args.
+    """
+    def __init__(self, cc=None, cxx=None, **kwargs):
```

Comment:
Sorry if I'm not picking up on something here, but why provide default arguments for arguments the documentation claims are necessary? If you're looking for the ability to use named arguments like `Toolchain(cc="foo", cxx="bar")`, you can do that without supplying default arguments in Python.